### PR TITLE
chore: add trivy vulnerability scanning

### PR DESCRIPTION
**Pull request by deansheather**
- Rename the `codeql` workflow to `security`
- Adds trivy vulnerability scanning for the linux_amd64 Docker image

### CODE REVIEW

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          restore-keys: |
+            js-${{ runner.os }}-
+
+      - name: Build Coder linux amd64 Docker image
+        id: build
+        run: |
+          set -euo pipefail
+          image_job="build/coder_$(./scripts/version.sh)_linux_amd64.tag"
+          make -j "$image_job"
+          echo "image=$(cat "$image_job")" >> $GITHUB_OUTPUT
```
Comment (mafredri) [inline .github/workflows/security.yml:110]: This is one of those "useless use of cat" 😄, consider an alternative:
```suggestion
          echo "image=$(<"$image_job")" >> $GITHUB_OUTPUT
```

File: .github/workflows/security.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@7b7aa264d83dc58691451798b4d117d53d21edfe
```
Comment (mafredri) [inline .github/workflows/security.yml:113]: Is this the recommended way (to lock down the sha)?

**Review by mafredri — APPROVED** [review]
Minor comments but looks good 👍🏻.

File: .github/workflows/security.yml (hunk 2 above)
Comment (deansheather) [reply .github/workflows/security.yml:119]: They recommend `@master` which I thought was a no-no. The autogenerated template that GitHub makes used a SHA but it wasn't latest so I used latest SHA instead.

File: .github/workflows/security.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+permissions:
+  actions: read
+  contents: read
+  security-events: write
+
+on:
+  push:
+    branches: ["main"]
+
+  pull_request:
```
[the remark this answers is not in the document]
Comment (deansheather) [reply .github/workflows/security.yml:12]: The scripts themselves that these security tasks run could read the token and write security events so using `pull_request_target` won't provide much protection :/

File: .github/workflows/security.yml (hunk 1 above)
Comment (deansheather) [reply .github/workflows/security.yml:116]: I like cat :(

**Pull request merged**